scripts/build_connectome/connectivity_plots.py

```diff
@@ -0,0 +1,854 @@
+        d["p_value"].extend([p_value for _ in range(size)])
+
+    return d
+
+
+def plot_summary_box(
+    df,
+    x,
+    y,
+    hue,
+    xlabel,
+    ylabel,
+    fig_title,
+    plot_filename,
+    mean_center,
+    fig_size=(10, 10),
+):
+    """Plot summary boxplot"""
+    plt.figure(figsize=fig_size)
+    sns.boxplot(df, orient="h", x=x, y=y, hue=hue, fliersize=0)
+    if mean_center:
+        xlabel = xlabel + " (double mean-centered)"
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(fig_title)
+    save_folder = "boxplots"
+    if mean_center:
+        save_folder += "_mean_centered"
+    if not os.path.exists(save_folder):
+        os.makedirs(save_folder)
+    plt.savefig(f"{save_folder}/{plot_filename}.png", bbox_inches="tight")
+    plt.close("all")
+
+
+def run_all_comparisons(
+    data_root,
+    tmp_dir,
+    task_pair,
+    func_conn_measures,
+    struc_conn_measures,
+    atlas,
+    d,
+    mean_center,
+):
+    """Manages specific choices related to calculation of pearson correlation
+    between two connectivity matrices, depending on the modality and the type
+    of connectivity measure
+
+    Parameters
+    ----------
+    data_root : str
+        path to the root directory of the data
+    tmp_dir : str
+        path to the temporary directory
+    task_pair : tuple of str
+        pair of tasks to be compared
+    func_conn_measures : list of str
+        calculated functional connectivity measures
+    struc_conn_measures : list of str
+        calculated structural connectivity measures
+    atlas : sklearn.utils.Bunch
+        Dictionary-like object, contains:
+        - 'maps': `str`, path to nifti file containing the
+        3D ~nibabel.nifti1.Nifti1Image (its shape is (182, 218, 182)). The
+        values are consecutive integers between 0 and n_rois which can be
+        interpreted as indices in the list of labels.
+        - 'labels': numpy.ndarray of str, array containing the ROI labels
+        including Yeo-network annotation.
+        - 'description': `str`, short description of the atlas
+          and some references.
+    d : dict
+        dictionary containing every single correlation value in a long-form
+        fashion (every row contains a single correlation in the correlation
+        column), and the corresponding information about the tasks,
+        the modality, the connectivity measure, and the comparison (same
+        subject vs cross subject)
+    mean_center : bool
+        whether to mean-center the correlation matrices
+
+    Returns
+    -------
+    dictionary
+        dictionary d defined above is updated with the new values
+    """
+    # get session numbers and modality for each task in the pair
+    task1, task2 = task_pair[0], task_pair[1]
+    task1_sub_ses, task1_modality = get_ses_modality(task1)
+    task2_sub_ses, task2_modality = get_ses_modality(task2)
+    combinations_plotted = []
+    if task1_modality == task2_modality == "functional":
```

Comment:
Can you avoid this very long if... else statements, e.g. y breaking them into subfunctions.